src/main/kotlin/app/revanced/patches/twitter/misc/links/ChangeLinkSharingDomainPatch.kt

```diff
@@ -0,0 +1,140 @@
+        required = true
+    )
+
+    override fun execute(context: BytecodeContext) {
+        val linkBuilderResult = LinkBuilderFingerprint.result ?: throw LinkBuilderFingerprint.exception
+
+        // region Copy link button.
+        linkBuilderResult.mutableMethod.apply {
+            val stringIndex = linkBuilderResult.scanResult.stringsScanResult!!.matches
+                .first().index
+
+            val targetRegister = getInstruction<OneRegisterInstruction>(stringIndex).registerA
+            replaceInstruction(
+                stringIndex,
+                "const-string v$targetRegister, \"https://$domainName/%1\$s/status/%2\$d\""
+            )
+        }
+
+
+        // endregion
+
+        // Used in the Share via... dialog.
+        LinkResourceGetterFingerprint.result?.apply {
+            val instructions = mutableMethod.getInstructions()
+
+            // Result register of the original method call.
+            var resultRegister = 0
+
+            // Remove instructions at the end and get the register of the last instruction.
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Landroid/content/res/Resources;") continue
+
+                    // Get the result register of the original method call.
+                    resultRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+
+                    mutableMethod.apply {
+                        // Remove the original method call for getting the link from resources and the move-result-object instruction.
+                        removeInstructions(index, 2)
+
+                        // Remove the instruction that uses the resultRegister as an array reference to prevent an error.
+                        removeInstructions(index - 2, 1)
+                    }
+
+                    break
+                }
+            }
+
+            // Instruction that sets free register to "this", needed to restore the original value of the register.
+            var instructionToGetThis: TwoRegisterInstruction? = null
+
+            // Save user nickname to free register.
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+                    val methodReference = (instruction as ReferenceInstruction).reference.toString()
+                    if (!methodReference.endsWith("Ljava/lang/String;")) continue
+
+                    // Get instruction that sets free register to "this".
+                    instructionToGetThis = mutableMethod.getInstruction<TwoRegisterInstruction>(index - 1)
+                    mutableMethod.addInstruction(
+                        index + 2,
```

Comment:
Name offsets like this, otherwise it is unclear where they point to without reading the apps source code